crates/transaction-pool/src/traits.rs

```diff
@@ -956,7 +956,7 @@ pub trait PoolTransaction: fmt::Debug + Send + Sync + Clone {
     /// For legacy transactions: `gas_price * gas_limit + tx_value`.
     /// For EIP-4844 blob transactions: `max_fee_per_gas * gas_limit + tx_value +
     /// max_blob_fee_per_gas * blob_gas_used`.
-    fn cost(&self) -> U256;
+    fn cost(&self) -> &U256;
```

Comment:
these make sense, this enforces a constraint that the tx must have this as field but this is actually good because this should always be stored